ext/semian/semian.c

```diff
@@ -73,9 +73,62 @@ void Init_semian()
   }
 }
 
+static const char*
+get_hostname()
+{
+  const char *VARIABLES[] = {
+    "KUBE_HOSTNAME",
+    "KUBE_HOST_NAME",
+    "KUBE_NODENAME",
+    "KUBE_NODE_NAME",
+    "NODENAME",
+    "NODE_NAME",
+    "HOSTNAME",
+    "HOST_NAME",
+  };
+
+  for (size_t i = 0; i < sizeof(VARIABLES); ++i) {
+    const char *hostname = getenv(VARIABLES[i]);
+    if (hostname != NULL) {
+      dprintf("Host name is %s", hostname);
+      return hostname;
+    }
+  }
+
+  return NULL;
+}
+
+static int
+force_c_circuits()
+{
+  const char *force_host = getenv("SEMIAN_CIRCUIT_BREAKER_FORCE_HOST");
+  if (!force_host) return 0;
+  const char *hostname = get_hostname();
+  if (!hostname) return 0;
+
+  int retval = 0;
+
+  char *hostnames = strdup(force_host);
+
+  char *cursor = NULL;
+  char *saveptr = hostnames;
+  while ((cursor = strtok_r(saveptr, ",; ", &saveptr))) {
+    if (strcmp(hostname, cursor) == 0) {
+      fprintf(stderr, "Info: Semian forcing host-based circuit implementation\n");
+      retval = 1;
+      break;
+    }
+  }
+
+  free(hostnames);
+  return retval;
+}
```

Comment:
My own preference would be for more of this logic to be in Ruby land (more familiar, easier to test, etc), and we'd swap out the implementations there, but this works ;)

Reply:
Me too, but this code is called when the module is being loaded and I'm not confident what Ruby code is available at this point.

Reply:
:+1: We can always come back and factor it out to Ruby land later. If it can be the same behaviourally, it shouldn't be an issue.

Reply:
I also agree with doing this in Ruby. I'm fairly sure the C extension is loaded here: https://github.com/Shopify/semian/blob/056cd1092d071ba5a970c8fc646e5525715fba77/lib/semian.rb#L270

Reply:
Done.